.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,32 @@
+name: Tag repo; build and publish assets
+on:
+  create:
+    tags: ['v*']
+  workspace_dispatch:
+
+jobs:
+  build-and-publish-docker-image:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    env:
+      image: ghcr.io/opensafely-core/cohortextractor-v2:latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Get current git tag
+        id: tags
+        run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
+
+      - name: Build image
+        run: docker build . --file Dockerfile --tag ${{ env.image }}
+
+      - name: Log into GitHub Container Registry
+        run: docker login https://ghcr.io -u ${{ github.actor }} --password ${{ secrets.DOCKER_RW_TOKEN }}
+
+      - name: Push image to GitHub Container Registry
+        env:
+          VERSION: ${{ steps.tags.outputs.tag }}
+        run: |
+          docker tag ${{ env.image }} ${{ env.image }}:latest
+          docker tag ${{ env.image }} ${{ env.image }}:$VERSION
+          docker push ${{ env.image }}:$VERSION
```

Comment:
Does this automatically change `:latest`? on the registry? I guess it should, but we don't know until we try!

Reply:
Time to find out! 🤞